indy_node/test/upgrade/test_broken_connection_control_tool.py

```diff
@@ -0,0 +1,69 @@
+import dateutil
+import dateutil.tz
+import pytest
+
+from datetime import datetime, timedelta
+from copy import deepcopy
+from indy_common.constants import START
+from indy_node.test.upgrade.conftest import patch_packet_mgr_output, EXT_PKT_NAME, EXT_PKT_VERSION
+
+from indy_node.server.upgrader import Upgrader
+from indy_node.test.upgrade.helper import sdk_ensure_upgrade_sent, bumpedVersion
+from plenum.test.helper import randomText
+
+from stp_core.common.log import getlogger
+
+delta = 2
+logger = getlogger()
+
+
+@pytest.fixture(scope="module")
+def tconf(tconf):
+    old_delta = tconf.MinSepBetweenNodeUpgrades
+    tconf.MinSepBetweenNodeUpgrades = delta
+    yield tconf
+    tconf.MinSepBetweenNodeUpgrades = old_delta
+
+
+@pytest.fixture(scope='function')
+def validUpgrade(nodeIds, tconf, monkeypatch):
+    schedule = {}
+    unow = datetime.utcnow().replace(tzinfo=dateutil.tz.tzutc())
+    startAt = unow + timedelta(seconds=delta)
+    for i in nodeIds:
+        schedule[i] = datetime.isoformat(startAt)
+        startAt = startAt + timedelta(seconds=delta)
+
+    patch_packet_mgr_output(monkeypatch, EXT_PKT_NAME, EXT_PKT_VERSION)
+
+    return dict(name='upgrade-{}'.format(randomText(3)), version=bumpedVersion(EXT_PKT_VERSION),
+                action=START, schedule=schedule, timeout=1, package=EXT_PKT_NAME,
+                sha256='db34a72a90d026dae49c3b3f0436c8d3963476c77468ad955845a1ccf7b03f55')
+
+
+@pytest.fixture(scope='function')
+def skip_functions():
+    # Do this to prevent exceptions because of node_control_tool absence
+    old_action_failed = deepcopy(Upgrader._action_failed)
+
+    Upgrader._action_failed = \
+        lambda self, version, scheduled_on, upgrade_id, external_reason: 1
+    yield
+    Upgrader._action_failed = old_action_failed
+
+
+def test_node_doesnt_retry_upgrade(looper, nodeSet, validUpgrade, nodeIds,
+                                   sdk_pool_handle, sdk_wallet_trustee, tconf,
+                                   skip_functions):
+    # Emulating connection problems
+    for node in nodeSet:
+        node.upgrader.retry_limit = 0
+
+    # Send upgrade
+    sdk_ensure_upgrade_sent(looper, sdk_pool_handle,
+                            sdk_wallet_trustee, validUpgrade)
+    looper.runFor(len(nodeIds) * delta)
+
```

Comment:
Can we check that Upgrade actually didn't happen in this case, that is we are still on the same version?